workspace_host/interface.js

```diff
@@ -957,7 +957,13 @@ async function sendLogs(workspaceId, res) {
   try {
     const workspace = await _getWorkspaceAsync(workspaceId);
     const container = await _getDockerContainerByLaunchUuid(workspace.launch_uuid);
-    const logs = await container.logs({ stdout: true, stderr: true, timestamps: true });
+    const logs = await container.logs({
+      stdout: true,
+      stderr: true,
+      timestamps: true,
+      // This should give us a reasonable bound on the worst-case performance.
+      tail: 50000,
```

Comment:
Good catch! I need to get in the habit of always adding limits to calls like this. Basically any call that's going to process data, tbh.